code-challenges.test.js

```diff
@@ -14,38 +14,149 @@
-const hodgepodge2 = [5, "Hola", 43, -34, "greetings", true]
-// Expected output: [ 2, 1, -1 ]
+describe("onlyNumbers", () => {
+  it("takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3", () => {
+    const hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false]
+    // Expected output: [ 2, 0, -1, 0 ]
+    const hodgepodge2 = [5, "Hola", 43, -34, "greetings", true]
+    // Expected output: [ 2, 1, -1 ]
+    expect(onlyNumbers(hodgepodge1)).toEqual([2, 0, -1, 0])
+    expect(onlyNumbers(hodgepodge2)).toEqual([2, 1, -1])
+  })
+})
+
+    // JEST "GOOD" FAILURE
+
+    //  ● onlyNumbers › takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3
+
+    // ReferenceError: onlyNumbers is not defined
 
 
 // b) Create the function that makes the test pass.
 
+    // Fuction ONLYNUMBERS takes in ARRAY
+    // FILTER over the ARRAY elements -> Condition? TYPEOF === NUMBER     <<<< set as variable getNumbers
+    // MAP over newArray elements and %= 3 on each     <<<< set as variable remainders
+    // RETURN remainders
+
+const onlyNumbers = (array) => {
+  let getNumbers = array.filter(element => typeof element === 'number')
+  let remainders = getNumbers.map(element => element %= 3)
+  return remainders
+}
+
+    // JEST TEST PASS
+
+    //  PASS  ./code-challenges.test.js
+
+    // onlyNumbers
+    
+    //     ✓ takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3(1 ms)
+
+
+
+
 
 // --------------------3) Create a function that takes in an array of numbers and returns the sum of all the numbers cubed.
 
 // a) Create a test with an expect statement using the variables provided.
 
-const cubeAndSum1 = [2, 3, 4]
-// Expected output: 99
-const cubeAndSum2 = [0, 5, 10]
-// Expected output: 1125
+describe("sumCubed", () => {
+  it("takes in an array of numbers and returns the sum of all the numbers cubed", () => {
+    const cubeAndSum1 = [2, 3, 4]
+    // Expected output: 99
+    const cubeAndSum2 = [0, 5, 10]
+    // Expected output: 1125
+    expect(sumCubed(cubeAndSum1)).toEqual(99)
+    expect(sumCubed(cubeAndSum2)).toEqual(1125)
+  })
+})
+
+    // JEST "GOOD" FAILURE
+
+    //   ● sumCubed › takes in an array of numbers and returns the sum of all the numbers cubed
+
+    // ReferenceError: sumCubed is not defined
 
 
 // b) Create the function that makes the test pass.
+
+    // FUNCTION sumCubed takes in ARRAY
+    // MAP over each Array Elemenmt and ** 3 on each.     <<<< set as Variable "cubed"
+    // Create "sum" variable to hold the value of "cubed".REDUCE (total, num)
+    // should be able to return the total PLUS num which came with the .reduce method
+    // finally return sum
+
+const sumCubed = (array) => {
+  let cubed = array.map(element => element ** 3)
+  let sum = cubed.reduce(function (total, num) {
+    return total + num;
+  })
+  return sum
+}
```

Comment:
Nice!  This could also be reduced down by chaining reduce and returning the same line.